crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -366,6 +366,10 @@ impl<'a> Resolver<'a> {
             }
         }
 
+        if let Some(type_alias_type) = self.lookup_type_alias(path.clone()) {
+            return type_alias_type;
```

Comment:
For future reference: this is likely where we'll need to apply generics once those are implemented.